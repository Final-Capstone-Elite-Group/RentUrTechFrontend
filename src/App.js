import React from 'react';
import './App.scss';

const App = () => (
  <div className="App">
    Hello
  </div>
);

export default App;
