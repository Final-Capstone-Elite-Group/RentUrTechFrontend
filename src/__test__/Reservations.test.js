/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from 'react';
import { setupServer } from 'msw/node';
import {
  waitFor, render, fireEvent, screen, cleanup, act,
} from './test-utils';
import App from '../App';
import '@testing-library/jest-dom';
import handlers from './mock/handlers';

const server = setupServer(...handlers);

jest.mock('react-select', () => ({ options = [{ label: 'tech', value: 1 }], value, onChange }) => {
  const handleChange = () => {
    onChange(1);
  };
  return (
    <select data-testid="select" value={value} onChange={handleChange}>
      {options.map(({ label, value }) => (
        <option key={value} value={value}>
          {label}
        </option>
      ))}
    </select>
  );
});

afterEach(() => cleanup(), server.resetHandlers());

beforeAll(() => server.listen({
  onUnhandledRequest: 'error',
}));

beforeEach(() => {
  jest.spyOn(console, 'error').mockImplementation(() => {});
});

let preloadedState;

beforeEach(() => {
  preloadedState = {
    auth: {
      user: {
        id: 25,
        name: 'string',
        username: 'unique string',
        email: 'test@example.com',
        role: 'admin',
      },
      token: '123',
    },
    reservation:
       [
         {
           id: 15,
           total: '52.0',
           reserved_date: '2022-07-26',
           city: 'Somewhere nice',
           created_at: '2022-06-09T15:07:15.493Z',
           updated_at: '2022-06-09T15:07:15.493Z',
           equipment: {
             id: 14,
             title: 'Wireless Printer',
             url: 'http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBGdz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--9d122d8fb7af2f04562aa6a26d728ed7d3fe8c64/drone1.jpg',
             duration: 5,
           },
         },
       ],
    equipment: {
      equipments: [
        {
          id: 14,
          title: 'Wireless Printer',
          description: "The Wireless Printer gives you all the essential features you needâ€”easily print, scan and copy, and activate HP+: HP's smart printing system that comes with 6 months of free Instant Ink and a 2 year extended HP warranty.",
          review: 'https://www.youtube.com/watch?v=9Xk39LNg304',
          duration: 5,
          rent_fee: '821.0',
          dates_reserved: [],
          total_amount_payable: '22.0',
          created_at: '2022-06-08T19:42:08.103Z',
          updated_at: '2022-06-08T19:42:08.112Z',
          image: {
            url: 'http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBJUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--2c533787f3499541cdd2bd6da005bace4749ddb7/wireless_printer.jpeg',
          },
        },
        {
          id: 13,
          title: 'Turntable',
          description: 'Whether you\'re just getting into records, rediscovering a record collection long since buried away or simply adding a turntable to complete your system, the AT-LP60X fully automatic belt-drive turntable is an excellent choice. This affordable, easy-to-use turntable plays both 33-1/3 and 45 RPM records and, with its built-in switchable phonon preamp, can be connected directly to your home stereo, powered speakers, computer and other components, whether they have a dedicated phonon input or not. An update of the popular AT-LP60, the ATLP60X features a redesigned tone arm base and head shell to improve tracking and reduce resonance. Noise is further reduced by the inclusion of an AC adapter that moves the AC/DC conVersion outside of the chassis where its impact on the signal chain is limited. Audio-Technical has been a leader in phonon cartridge design for more than 50 years, and that expertise shows in the AT-LP60X\'s Dual Magnet cartridge with replaceable stylus. The cartridge is integrated into the head shell for assured performance and ease of setup. The turntable comes with a detachable RCA output cable (3. 5 mm male to dual RCA male), 45 RPM adapter, and a removable hinged dust cover. It is available in black (AT-LP60X-BK), brown/black (AT-LP60X-BW), gunmetal/black (AT-LP60X-GM), and red/black (AT-LP60X-RD).',
          review: 'https://www.youtube.com/watch?v=Yaj1Mhj8mo4',
          duration: 6,
          rent_fee: '832.0',
          dates_reserved: [],
          total_amount_payable: '26.0',
          created_at: '2022-06-08T19:42:08.067Z',
          updated_at: '2022-06-08T19:42:08.081Z',
          image: {
            url: 'http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBJQT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ee3004a8fe9fce926a806f65e4a87b9aa25c45af/turntable.jpeg',
          },
        }],
      currentTech: {
        id: 13,
        title: 'Turntable',
        description: 'Whether you\'re just getting into records, rediscovering a record collection long since buried away or simply adding a turntable to complete your system, the AT-LP60X fully automatic belt-drive turntable is an excellent choice. This affordable, easy-to-use turntable plays both 33-1/3 and 45 RPM records and, with its built-in switchable phonon preamp, can be connected directly to your home stereo, powered speakers, computer and other components, whether they have a dedicated phonon input or not. An update of the popular AT-LP60, the ATLP60X features a redesigned tone arm base and head shell to improve tracking and reduce resonance. Noise is further reduced by the inclusion of an AC adapter that moves the AC/DC conVersion outside of the chassis where its impact on the signal chain is limited. Audio-Technical has been a leader in phonon cartridge design for more than 50 years, and that expertise shows in the AT-LP60X\'s Dual Magnet cartridge with replaceable stylus. The cartridge is integrated into the head shell for assured performance and ease of setup. The turntable comes with a detachable RCA output cable (3. 5 mm male to dual RCA male), 45 RPM adapter, and a removable hinged dust cover. It is available in black (AT-LP60X-BK), brown/black (AT-LP60X-BW), gunmetal/black (AT-LP60X-GM), and red/black (AT-LP60X-RD).',
        review: 'https://www.youtube.com/watch?v=Yaj1Mhj8mo4',
        duration: 6,
        rent_fee: '832.0',
        dates_reserved: ['2022-12-12'],
        total_amount_payable: '26.0',
        created_at: '2022-06-08T19:42:08.067Z',
        updated_at: '2022-06-08T19:42:08.081Z',
        image: {
          url: 'http://localhost:3000/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBJQT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ee3004a8fe9fce926a806f65e4a87b9aa25c45af/turntable.jpeg',
        },

      },
    },
  };
});

afterAll(() => server.close());

it('App Renders Correctly and takes you to Tech page', async () => {
  const { container } = render(<App />);
  await waitFor(() => screen.getByText('Wireless Printer'));
  expect(container).toMatchSnapshot();
});

it('Clicking on reservation takes you to the reservations page', async () => {
  render(<App />, { preloadedState });
  await waitFor(() => screen.getByText('Wireless Printer'));
  fireEvent.click(screen.getByText('Reservation'));
  await waitFor(() => screen.getByText('Book Now'));
  expect(screen.getByText('Book Now')).toBeInTheDocument();
});

it('Clicking on Add Reservation without values gives error', async () => {
  render(<App />, { preloadedState });
  await waitFor(() => screen.getByText('Book Now'));
  fireEvent.click(screen.getByTestId('submitReservation'));
  await waitFor(() => screen.getByText('Please make sure you fill in all fields'));
  expect(screen.getByText('Please make sure you fill in all fields')).toBeInTheDocument();
});

it('Clicking on Add Reservation with values does not give error', async () => {
  render(<App />, { preloadedState });
  await act(async () => {
    await waitFor(() => screen.getByText('Book Now'));
    fireEvent.change(screen.getAllByTestId('select')[0], { target: { value: 0 } });
    fireEvent.change(screen.getAllByTestId('select')[1], { target: { value: 1 } });
    fireEvent.change(screen.getByRole('textbox'), { target: { value: '07/08/2022' } });
    fireEvent.click(screen.getByTestId('submitReservation'));
    const error = screen.queryByText('Please make sure you fill in all fields');
    expect(error).not.toBeInTheDocument();
  });
});

it('Clicking on My Reservations takes you to the your Reservations page', async () => {
  render(<App />, { preloadedState });
  fireEvent.click(screen.getByTestId('hamburger'));
  fireEvent.click(screen.getByText('My Reservations'));
  await waitFor(() => screen.getByText('Active Reservations'));
  expect(screen.getByText('Active Reservations')).toBeInTheDocument();
});

it('Clicking on Trashcan deletes the Reservation', async () => {
  render(<App />, { preloadedState });
  await waitFor(() => screen.getByText('Wireless Printer'));
  fireEvent.click(screen.getByTestId('trashcan'));
  screen.getByText('Wireless Printer');
  await waitFor(() => expect(screen.queryByText('Wireless Printer')).not.toBeInTheDocument());
});
