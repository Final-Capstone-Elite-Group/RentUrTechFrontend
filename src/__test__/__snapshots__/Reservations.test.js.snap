// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly and takes you to Tech page 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="hamburger"
    >
      <svg
        data-testid="hamburger"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 7a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 13a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <aside
      class="relative false"
    >
      <button
        class="closing_button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
          />
        </svg>
      </button>
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="main_logo.gif"
        />
      </div>
      <nav>
        <ul
          class="nav_list"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Techs
            </a>
          </li>
          <li>
            <a
              class="inactive"
              href="/reservation"
            >
              Reservation
            </a>
          </li>
          <li>
            <a
              class="inactive"
              href="/my-reservations"
            >
              My Reservations
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="social_links"
      >
        <div
          class="user_login"
        >
          <a
            class=""
            href="/login"
          >
            Login
          </a>
          <a
            class=""
            href="/signup"
          >
            Sign up
          </a>
        </div>
        <ul>
          <li>
            <a
              aria-label="Save"
              href="https://twitter.com/?lang=en"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.026 15c6.038 0 9.341-5.003 9.341-9.334 0-.14 0-.282-.006-.422A6.685 6.685 0 0 0 16 3.542a6.658 6.658 0 0 1-1.889.518 3.301 3.301 0 0 0 1.447-1.817 6.533 6.533 0 0 1-2.087.793A3.286 3.286 0 0 0 7.875 6.03a9.325 9.325 0 0 1-6.767-3.429 3.289 3.289 0 0 0 1.018 4.382A3.323 3.323 0 0 1 .64 6.575v.045a3.288 3.288 0 0 0 2.632 3.218 3.203 3.203 0 0 1-.865.115 3.23 3.23 0 0 1-.614-.057 3.283 3.283 0 0 0 3.067 2.277A6.588 6.588 0 0 1 .78 13.58a6.32 6.32 0 0 1-.78-.045A9.344 9.344 0 0 0 5.026 15z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-label="Save"
              href="https://twitter.com/?lang=en"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-label="Save"
              href="https://twitter.com/?lang=en"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M386.061 228.496c1.834 9.692 3.143 19.384 3.143 31.956C389.204 370.205 315.599 448 204.8 448c-106.084 0-192-85.915-192-192s85.916-192 192-192c51.864 0 95.083 18.859 128.611 50.292l-52.126 50.03c-14.145-13.621-39.028-29.599-76.485-29.599-65.484 0-118.92 54.221-118.92 121.277 0 67.056 53.436 121.277 118.92 121.277 75.961 0 104.513-54.745 108.965-82.773H204.8v-66.009h181.261zm185.406 6.437V179.2h-56.001v55.733h-55.733v56.001h55.733v55.733h56.001v-55.733H627.2v-56.001h-55.733z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-label="Save"
              href="https://twitter.com/?lang=en"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M447.8 153.6c-2 43.6-32.4 103.3-91.4 179.1-60.9 79.2-112.4 118.8-154.6 118.8-26.1 0-48.2-24.1-66.3-72.3C100.3 250 85.3 174.3 56.2 174.3c-3.4 0-15.1 7.1-35.2 21.1L0 168.2c51.6-45.3 100.9-95.7 131.8-98.5 34.9-3.4 56.3 20.5 64.4 71.5 28.7 181.5 41.4 208.9 93.6 126.7 18.7-29.6 28.8-52.1 30.2-67.6 4.8-45.9-35.8-42.8-63.3-31 22-72.1 64.1-107.1 126.2-105.1 45.8 1.2 67.5 31.1 64.9 89.4z"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-label="Save"
              href="https://twitter.com/?lang=en"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M204 6.5C101.4 6.5 0 74.9 0 185.6 0 256 39.6 296 63.6 296c9.9 0 15.6-27.6 15.6-35.4 0-9.3-23.7-29.1-23.7-67.8 0-80.4 61.2-137.4 140.4-137.4 68.1 0 118.5 38.7 118.5 109.8 0 53.1-21.3 152.7-90.3 152.7-24.9 0-46.2-18-46.2-43.8 0-37.8 26.4-74.4 26.4-113.4 0-66.2-93.9-54.2-93.9 25.8 0 16.8 2.1 35.4 9.6 50.7-13.8 59.4-42 147.9-42 209.1 0 18.9 2.7 37.5 4.5 56.4 3.4 3.8 1.7 3.4 6.9 1.5 50.4-69 48.6-82.5 71.4-172.8 12.3 23.4 44.1 36 69.3 36 106.2 0 153.9-103.5 153.9-196.8C384 71.3 298.2 6.5 204 6.5z"
                />
              </svg>
            </a>
          </li>
        </ul>
        <p
          class="copy"
        >
           © Copyright 2022, RUT incorporated
        </p>
      </div>
    </aside>
    <div
      class="Toastify"
    />
    <section
      class="tech_section"
    >
      <div
        class="tech_desc"
      >
        <h1>
          Latest Techs
        </h1>
        <p>
          Please Select a device to check it's details
        </p>
      </div>
      <div
        class="tech_carousel"
      >
        <div
          class="container"
          data-testid="carousel"
        >
          <button
            class="button "
            data-testid="buttonPrevious"
            disabled=""
            style="transform: translateY(-50%); position: absolute;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
              />
            </svg>
          </button>
          <div
            class="cardContainer"
          >
            <div
              class="card"
            >
              <div
                class="cardImageWrapper"
              >
                <div
                  class="imageBackground"
                >
                  <figure>
                    <img
                      alt="tool"
                      class="image"
                      src="https://renturtech.herokuapp.com/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBJUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--2c533787f3499541cdd2bd6da005bace4749ddb7/wireless_printer.jpeg"
                    />
                  </figure>
                </div>
              </div>
              <a
                class="linkToEquipment"
                href="/details/14"
              >
                <h2>
                  Wireless Printer
                </h2>
                <div
                  class="separator"
                >
                  ....................
                </div>
                <p
                  class="cardContent"
                >
                  The Wireless Printer gives you all the essential features you need—easily print, scan and copy, and activate HP+: HP's smart printing system that comes with 6 months of free Instant Ink and a 2 year extended HP warranty.
                </p>
              </a>
              <ul
                class="iconContainer"
              >
                <li
                  class="iconWrapper"
                >
                  <a
                    aria-label="link-to-social"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="iconWrapper"
                >
                  <a
                    aria-label="link-to-social"
                    href="#"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 0 1-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 0 1-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 0 0 229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="iconWrapper"
                >
                  <a
                    aria-label="link-to-social"
                    href="https://www.youtube.com/watch?v=9Xk39LNg304"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8.051 1.999h.089c.822.003 4.987.033 6.11.335a2.01 2.01 0 0 1 1.415 1.42c.101.38.172.883.22 1.402l.01.104.022.26.008.104c.065.914.073 1.77.074 1.957v.075c-.001.194-.01 1.108-.082 2.06l-.008.105-.009.104c-.05.572-.124 1.14-.235 1.558a2.007 2.007 0 0 1-1.415 1.42c-1.16.312-5.569.334-6.18.335h-.142c-.309 0-1.587-.006-2.927-.052l-.17-.006-.087-.004-.171-.007-.171-.007c-1.11-.049-2.167-.128-2.654-.26a2.007 2.007 0 0 1-1.415-1.419c-.111-.417-.185-.986-.235-1.558L.09 9.82l-.008-.104A31.4 31.4 0 0 1 0 7.68v-.123c.002-.215.01-.958.064-1.778l.007-.103.003-.052.008-.104.022-.26.01-.104c.048-.519.119-1.023.22-1.402a2.007 2.007 0 0 1 1.415-1.42c.487-.13 1.544-.21 2.654-.26l.17-.007.172-.006.086-.003.171-.007A99.788 99.788 0 0 1 7.858 2h.193zM6.4 5.209v4.818l4.157-2.408L6.4 5.209z"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="card"
            >
              <div
                class="cardImageWrapper"
              >
                <div
                  class="imageBackground"
                >
                  <figure>
                    <img
                      alt="tool"
                      class="image"
                      src="https://renturtech.herokuapp.com/rails/active_storage/blobs/redirect/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBJQT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ee3004a8fe9fce926a806f65e4a87b9aa25c45af/turntable.jpeg"
                    />
                  </figure>
                </div>
              </div>
              <a
                class="linkToEquipment"
                href="/details/13"
              >
                <h2>
                  Turntable
                </h2>
                <div
                  class="separator"
                >
                  ....................
                </div>
                <p
                  class="cardContent"
                >
                  Whether you’re just getting into records, rediscovering a record collection long since buried away or simply adding a turntable to complete your system, the AT-LP60X fully automatic belt-drive turntable is an excellent choice. This affordable, easy-to-use turntable plays both 33-1/3 and 45 RPM records and, with its built-in switchable phonon preamp, can be connected directly to your home stereo, powered speakers, computer and other components, whether they have a dedicated phonon input or not. An update of the popular AT-LP60, the ATLP60X features a redesigned tone arm base and head shell to improve tracking and reduce resonance. Noise is further reduced by the inclusion of an AC adapter that moves the AC/DC conVersion outside of the chassis where its impact on the signal chain is limited. Audio-Technical has been a leader in phonon cartridge design for more than 50 years, and that expertise shows in the AT-LP60X’s Dual Magnet cartridge with replaceable stylus. The cartridge is integrated into the head shell for assured performance and ease of setup. The turntable comes with a detachable RCA output cable (3. 5 mm male to dual RCA male), 45 RPM adapter, and a removable hinged dust cover. It is available in black (AT-LP60X-BK), brown/black (AT-LP60X-BW), gunmetal/black (AT-LP60X-GM), and red/black (AT-LP60X-RD).
                </p>
              </a>
              <ul
                class="iconContainer"
              >
                <li
                  class="iconWrapper"
                >
                  <a
                    aria-label="link-to-social"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="iconWrapper"
                >
                  <a
                    aria-label="link-to-social"
                    href="#"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 0 1-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 0 1-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 0 0 229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="iconWrapper"
                >
                  <a
                    aria-label="link-to-social"
                    href="https://www.youtube.com/watch?v=Yaj1Mhj8mo4"
                    rel="noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8.051 1.999h.089c.822.003 4.987.033 6.11.335a2.01 2.01 0 0 1 1.415 1.42c.101.38.172.883.22 1.402l.01.104.022.26.008.104c.065.914.073 1.77.074 1.957v.075c-.001.194-.01 1.108-.082 2.06l-.008.105-.009.104c-.05.572-.124 1.14-.235 1.558a2.007 2.007 0 0 1-1.415 1.42c-1.16.312-5.569.334-6.18.335h-.142c-.309 0-1.587-.006-2.927-.052l-.17-.006-.087-.004-.171-.007-.171-.007c-1.11-.049-2.167-.128-2.654-.26a2.007 2.007 0 0 1-1.415-1.419c-.111-.417-.185-.986-.235-1.558L.09 9.82l-.008-.104A31.4 31.4 0 0 1 0 7.68v-.123c.002-.215.01-.958.064-1.778l.007-.103.003-.052.008-.104.022-.26.01-.104c.048-.519.119-1.023.22-1.402a2.007 2.007 0 0 1 1.415-1.42c.487-.13 1.544-.21 2.654-.26l.17-.007.172-.006.086-.003.171-.007A99.788 99.788 0 0 1 7.858 2h.193zM6.4 5.209v4.818l4.157-2.408L6.4 5.209z"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <button
            class="button right"
            data-testid="buttonNext"
            disabled=""
            style="transform: translateY(-50%); position: absolute; right: 0px;"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
              />
            </svg>
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
